fhir-server/src/main/java/com/ibm/fhir/server/operation/spi/AbstractOperation.java

```diff
@@ -157,6 +168,45 @@ protected void validateOperationContext(FHIROperationContext operationContext, C
         }
     }
 
+    /**
+     * Determines if the operation disallows the GET method.
+     * This is determined by the affectsState value of the OperatorDefinition and whether the
+     * OperatorDefinition contains any non-primitive parameters.
+     * @param operationDefinition the operation definition
+     * @param parameters the parameters
+     * @return true or false
+     */
+    private boolean isGetMethodAllowed(OperationDefinition operationDefinition, Parameters parameters) {
```

Comment:
The OperationDefinitions probably don't need to be reprocessed for each request.  Maybe in this case it should be cached?  Since this is at the tail end of the expression, I have some doubts about the need.  

A cache feels ideal for the below... once warmed up, the cacheis sued.

Reply:
Since the passed in parameters on the request can be different on each request, they would still need to be checked each time, so I'm not sure what we'd be able to cache.

Reply:
you could minimally cache the OperationDefinition rather than process it each time. a point lookup after the processing of the OperationDefinition is then possible afterwards.